contracts/InsuranceArbitrator.sol

```diff
@@ -144,5 +153,39 @@ contract InsuranceArbitrator {
         uint256 timestamp,
         bytes memory ancillaryData,
         int256 price
-    ) external {}
+    ) external {
+        bytes32 claimId = _getClaimId(timestamp, ancillaryData);
+        require(address(insuranceClaims[claimId].optimisticOracle) == msg.sender, "Unauthorized callback");
+
+        // Claim can be settled only once, thus should be deleted.
+        bytes32 policyId = insuranceClaims[claimId].policyId;
+        InsurancePolicy storage claimedPolicy = insurancePolicies[policyId];
+        string memory insuredEvent = claimedPolicy.insuredEvent;
+        delete insuranceClaims[claimId];
+
+        address insuredAddress = claimedPolicy.insuredAddress;
+        IERC20 currency = claimedPolicy.currency;
+        uint256 insuredAmount = claimedPolicy.insuredAmount;
```

Comment:
can probs delete these declarations and just use the props directly from `claimedPolicy`.

Reply:
yes, I used them only for gas savings as they are referenced also in emitted logs. But to make contract more concise for demonstration purposes we can definitely skip this. After merging all implementations I would have separate PR for shortening the contract.